### PR TITLE
[pkg/otlp/metrics] Add support for metrics origin

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Adds support for setting metrics origin on metrics based on the scope.

### CODE REVIEW

**Comment by mx-psi** [comment]
@mackjmr requesting your review since you reviewed #204, but this one is less urgent IMO than #231

File: pkg/otlp/metrics/origin.go [hunk 1]
```diff
@@ -14,11 +14,198 @@
...
+	default:
+		return fmt.Sprintf("OriginCategory(%d)", o)
+	}
+}
+
 // OriginService defines the origin service.
 type OriginService int32
+
+// List all receivers that set the scope name.
+const (
```
Comment (mackjmr) [inline pkg/otlp/metrics/origin.go:66]: Missing `jmxreceiver` and `podmanreceiver` which both produce metrics.

Comment (mackjmr) [reply pkg/otlp/metrics/origin.go:66]: Ah nevermind, these two receivers don't set scope name.

**Review by mackjmr — APPROVED** [review]
A dedicated test in `metrics_translator_test.go` with something like:
```
	assert.ElementsMatch(t,
		consumer.metrics,
		[]metric{
			newCount(exampleDimsWithOrigin, uint64(seconds(1)), 30),
		},
	)

```

would be nice to have here, as this goes through the whole path. 

Other than that LGTM!

**Comment by mx-psi** [comment]
> A dedicated test in `metrics_translator_test.go` with something like:
> 
> ```
> 	assert.ElementsMatch(t,
> 		consumer.metrics,
> 		[]metric{
> 			newCount(exampleDimsWithOrigin, uint64(seconds(1)), 30),
> 		},
> 	)
> ```
> 
> would be nice to have here, as this goes through the whole path.
> 
> Other than that LGTM!

All the tests using the files in `pkg/otlp/metrics/testdata` are already testing this. I don't have one specific for a metrics product, maybe I can add one of those?

File: pkg/otlp/metrics/origin.go [hunk 2]
```diff
@@ -14,11 +14,198 @@
...
+
 // OriginProduct defines the origin product.
 type OriginProduct int32
 
+const (
+	OriginProductUnknown OriginProduct = 0
+	// OriginProductDatadogAgent is the origin for metrics coming from the Datadog Agent OTLP Ingest.
+	OriginProductDatadogAgent OriginProduct = 10
+	// OriginProductDatadogExporter is the origin for metrics coming from the OTel Collector Datadog Exporter.
+	OriginProductDatadogExporter OriginProduct = 17
```
Comment (mx-psi) [inline pkg/otlp/metrics/origin.go:30]: I am going to mark back as draft until we confirm this

**Comment by mackjmr** [comment]
> All the tests using the files in pkg/otlp/metrics/testdata are already testing this. I don't have one specific for a metrics product, maybe I can add one of those?

Yeah, I think this would be good. Also, not sure if it is possible, but if we can set the scope in one of the tests it would be nice to include a known origin service in there.

File: pkg/otlp/metrics/origin.go [hunk 3]
```diff
@@ -14,11 +14,198 @@
...
+		return "otlp"
+	default:
+		return fmt.Sprintf("OriginCategory(%d)", o)
+	}
+}
+
 // OriginService defines the origin service.
 type OriginService int32
+
+// List all receivers that set the scope name.
```
Comment (mackjmr) [inline pkg/otlp/metrics/origin.go:65]: I think this should also contain `connectors` ? A few add the scope: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/countconnector/connector.go#L79.

File: pkg/otlp/metrics/origin.go (hunk 2 above)
Comment (mx-psi) [reply pkg/otlp/metrics/origin.go]: Removing it for now

File: pkg/otlp/metrics/origin.go (hunk 3 above)
Comment (mx-psi) [reply pkg/otlp/metrics/origin.go:61]: Spoke offline about this, makes sense but will add it later since it needs to go through backend approval

**Comment by mx-psi** [comment]
Added a specific test for this, auto-merging....

**Pull request merged**